### PR TITLE
Deploy multi version docs to subdomain using pyansys/actions v4

### PR DESCRIPTION
**Pull request by ludovicsteinbach**
Modify CI to use doc deployment actions from pyansys/actions v4.
Bump version of sphinx theme to support multi-version
Configures sphinx for multi-version

Currently configured to use cname `bomanalytics.grantami.docs.pyansys.com` which hasn't been requested/verified yet

### CODE REVIEW

**Review by Andy-Grigg — APPROVED** [review]
Approved, assuming this PR won't do anything that might impact the current live pages without some other config changes

File: doc/source/conf.py
Comment (ludovicsteinbach) [reply doc/source/conf.py:139]: I'm not sure if publishing will work at all, because the cname is not available.

But it wouldn't affect existing docs, as they are still generated from the other repo

Comment (ludovicsteinbach) [reply doc/source/conf.py:139]: At this point we could request bomanalytics.grantami.docs.pyansys.com. For a while we'd have the bomanalytics served both at `bomanalytics.grantami` and `grantami`, but that would enable to sort things through before changing the landing page at `grantami.docs`

Comment (Andy-Grigg) [reply doc/source/conf.py:139]: That makes sense. We could even get to the point of having the bomanalytics docs live, and then only change the grantami page from bomanalytics to the Granta MI page when we want to release recordlists.

Sounds good to me. Yes, go ahead and request bomanalytics.grantami....

**Pull request merged**
